### DISCUSSION TITLE
Issues when deploying

### DISCUSSION THREAD

**Discussion by felixyeboah** [epicweb-dev/epic-stack#644] [Q&A] [answered]
![CleanShot 2024-03-09 at 22 18 15@2x](https://github.com/epicweb-dev/epic-stack/assets/87368189/f03cc039-023d-436b-80b0-837fc486065d)

I have this issue when I am deploying on fly.io. I have tried all I can but I don't know what's wrong. I have uninstalled and re-installed Faker and done anything and everything I can to make this work but nothing works. Any help, please? If anyone has faced this before.

**Comment by kentcdodds** [comment, marked as answer]
You shouldn't be importing anything from faker in production. It's hard for anyone to help you out without seeing the code. All I can tell you is what the error says. `@faker-js/faker` can't be found. This is because it's a development dependency. So you have some production code that's importing `@faker-js/faker` somehow.

You could run the build locally, then look at `./build/server/index.js` and find where `@faker-js/faker` is being imported and that may give you an idea of which module is importing it.

I hope that helps.

**Reply by felixyeboah** [reply]
yes, I am not importing or touching `@faker-js/faker` in any way. this is from the default bootstrapping of Epic Stack. I am not sure I am using it in any part of the application I worked on besides the usage in the `tests` folder. I will do well to check if I mistakenly also used it anywhere in my code base besides the `tests` folder and take it out.

**Reply by kentcdodds** [reply]
You must have something different from the default epic stack because its latest deployed version is working fine: https://github.com/epicweb-dev/epic-stack/actions/runs/8196931121/job/22418254777 https://www.epicstack.dev/

**Reply by felixyeboah** [reply]
![CleanShot 2024-03-09 at 23 41 51@2x](https://github.com/epicweb-dev/epic-stack/assets/87368189/b0ad3154-5cb7-4174-8271-80900922bbad)

after building, this is the only instance I could see` @faker-js/faker` being imported but it doesn't give me any idea on which module is importing it.

**Reply by felixyeboah** [reply]
it worked fine until I made a couple of changes in the UI which I am very sure I never touched `@faker-js/faker` in any of the routes or the components I created or worked on. I got this error when I was deploying the new changes.

**Reply by kentcdodds** [reply]
There's probably a tool you can use with vite to explore the module graph or something. Unfortunately I'm not sure what to recommend. I just checked the epic stack and I'm not getting that module included anywhere in the built file so I'm afraid there's not much more I can do to help. Sorry.

**Reply by kentcdodds** [reply]
> it worked fine until I made a couple of changes in the UI

It's possible the changes you made resulted in adding an import to a module which does import faker. That's most likely.

**Reply by felixyeboah** [reply]
thank you. you have given me a head start so I know how I can fix this issue or where to start looking. Thank you.

**Reply by felixyeboah** [reply]
@kentcdodds I figured the issue out and solved it but I have one question, please. how do you determine if a package should be built into the server file or not?

**Reply by kentcdodds** [reply]
It's it's only used by loaders or actions then I put it in a server file

**Reply by felixyeboah** [reply]
thank you for this
